Pull upstream changes & Ruby 3 compat

StoryparkNick opened the pull request:
Just to solve an issue that's preventing the Ruby 3 upgrade.

ivanovv committed:
Fixed an error when the text has a non-word character and a space next to it

ArturT committed:
Merge pull request #17 from ivanovv/fix-nil-error

Fixed an error when the text has a non-word character and a space next to it

ArturT committed:
Bump version and update changelog

peterzhu2118 committed:
Allow separator in options

peterzhu2118 committed:
Update readme

peterzhu2118 committed:
Update travis to test on newer Ruby verisons

ArturT committed:
Merge pull request #18 from peterzhu2118/custom-separator

Allow custom separator to be used in text

ArturT committed:
Update CHANGELOG.md

ArturT committed:
Add me to authors

ArturT committed:
Bump version 1.6.0

DavertMik committed:
Support for RMagick 4.1+

Recently I started receiving these deprecation messages:

```
gems/avatarly-1.6.0/lib/avatarly.rb:51: warning: passing a block without an image argument is deprecated
```

Seems like `self.` should not be used from within a block. This patch solved the issue for me.

ArturT committed:
Merge pull request #21 from DavertMik/patch-1

Support for RMagick 4.1+

ArturT committed:
Update CHANGELOG.md

ArturT committed:
Bump version 1.6.1

ArturT committed:
fix avatarly demo image

ArturT committed:
Add  Maintainers and support

ArturT committed:
Update README.md

ArturT committed:
Update README.md

ArturT committed:
Merge pull request #22 from ArturT/maintenance

Add maintainers and support section

StoryparkNick committed:
Merge branch 'master' of https://github.com/lucek/avatarly into pull-upstream

fourseven reviewed (approved):
Yup, assuming ruby 3.0 is happy with this

StoryparkNick committed:
Pass opts through to initials

StoryparkNick committed:
Set blob quality and depth via arg

Doing it via self seems to try and set it on the Avatarly class in Ruby
3, which doesn't work out very well.

StoryparkNick committed:
Update specs and fixture images

fourseven commented:
Still good, merge away
